rosette_apiUnitTests/rosette_apiUnitTests.cs

```diff
@@ -1156,46 +1156,104 @@ public void TextEmbeddingTestFull()
+            _mockHttp.When(_testUrl + "semantics/vector")
                 .Respond(HttpStatusCode.OK, "application/json", "{'response': 'OK'}");
 
-            var response = _rosetteApi.TextEmbedding("content");
+            var response = _rosetteApi.SemanticVectors("content");
 # pragma warning disable 618
             Assert.AreEqual(response.Content["response"], "OK");
 # pragma warning restore 618
         }
 
         [Test]
-        public void TextEmbedding_Dict_Test()
+        public void SemanticVectors_Dict_Test()
         {
-            _mockHttp.When(_testUrl + "text-embedding")
+            _mockHttp.When(_testUrl + "semantics/vector")
                 .Respond(HttpStatusCode.OK, "application/json", "{'response': 'OK'}");
 
-            var response = _rosetteApi.TextEmbedding(new Dictionary<object, object>() { { "contentUri", "contentUrl" } });
+            var response = _rosetteApi.SemanticVectors(new Dictionary<object, object>() { { "contentUri", "contentUrl" } });
 # pragma warning disable 618
             Assert.AreEqual(response.Content["response"], "OK");
 # pragma warning restore 618
         }
 
         [Test]
-        public void TextEmbedding_File_Test()
+        public void SemanticVectors_File_Test()
         {
-            _mockHttp.When(_testUrl + "text-embedding")
+            _mockHttp.When(_testUrl + "semantics/vector")
                 .Respond("application/json", "{'response': 'OK'}");
 
             RosetteFile f = new RosetteFile(_tmpFile);
-            var response = _rosetteApi.TextEmbedding(f);
+            var response = _rosetteApi.SemanticVectors(f);
+# pragma warning disable 618
+            Assert.AreEqual(response.Content["response"], "OK");
+# pragma warning restore 618
+        }
+
+        //------------------------- Similar Terms ----------------------------------------
+        [Test]
+        public void SimilarTermsTestFull()
+        {
+            Init();
+            IDictionary<string, List<SimilarTerm>> terms = new Dictionary<string, List<SimilarTerm>>() {
+                {"eng", new List<SimilarTerm>() {new SimilarTerm("spy", 1.0m)}}
+            };
+            string headersAsString = " { \"Content-Type\": \"application/json\", \"date\": \"Thu, 11 Aug 2016 15:47:32 GMT\", \"server\": \"openresty\", \"strict-transport-security\": \"max-age=63072000; includeSubdomains; preload\", \"x-rosetteapi-app-id\": \"1409611723442\", \"x-rosetteapi-concurrency\": \"50\", \"x-rosetteapi-request-id\": \"d4176692-4f14-42d7-8c26-4b2d8f7ff049\", \"content-length\": \"72\", \"connection\": \"Close\" }";
+            Dictionary<string, string> responseHeaders = new JavaScriptSerializer().Deserialize<Dictionary<string, string>>(headersAsString);
+            Dictionary<string, object> content = new Dictionary<string, object>() {
+                {"similarTerms", terms}
+            };
+            SimilarTermsResponse expected = new SimilarTermsResponse(terms, responseHeaders, content, null);
```

Comment:
Ah, @iredpath I think content needs to be a `string` not `Dictionary` here.

Reply:
Assuming I'm not reading the assert error response backwards (which would make this an even more confusing error), I think the actual issue is with the mocked response.  Even though I'm fairly sure I checked that `mockedContent` looks correct, by the time it comes back in the mocked response we lose the inner `terms` map.  I don't know if the issue is the `mockedMessage` itself or the mocked response as a whole, but maybe a (de)serialization issue?

Reply:
Sorry my eyes were crossed. I'm more noob than you when comes to c#. There's on other suspect:

https://github.com/rosette-api/csharp/blob/12444107b7256a1d7ba4d10c619d1066fba84e31/rosette_api/SimilarTermsResponse.cs#L45-L46

would the default `content = null` wipe out something?

Reply:
Doesn't look like it, I just gave it a shot and no change in the job.